### PR TITLE
RCAL-941: Update romancal to use new L1/L2 schema.

### PR DESCRIPTION
**Pull request by mairanteodoro**
Resolves [RCAL-941](https://jira.stsci.edu/browse/RCAL-941)

This PR updates `romancal` to comply with the changes in L1/L2 schema. All the unit and regression tests have been updated.

## Regression Tests
- https://github.com/spacetelescope/RegressionTests/actions/runs/11579151910

## Tasks
- [X] **request a review from someone specific**, to avoid making the maintainers review every PR
- [X] add a build milestone, i.e. `24Q4_B15` (use the [latest build](https://github.com/spacetelescope/romancal/milestones) if not sure)
- [ ] Does this PR change user-facing code / API? (if not, label with `no-changelog-entry-needed`)
  - [x] write news fragment(s) in `changes/`: `echo "changed something" > changes/<PR#>.<changetype>.rst` (see below for change types)
  - [X] update or add relevant tests
  - [ ] update relevant docstrings and / or `docs/` page
  - [X] [start a regression test](https://github.com/spacetelescope/RegressionTests/actions/workflows/romancal.yml) and include a link to the running job ([click here for instructions](https://github.com/spacetelescope/RegressionTests/blob/main/docs/running_regression_tests.md))
    - [ ] Do truth files need to be updated ("okified")?
      - [ ] **after the reviewer has approved these changes**, run `okify_regtests` to update the truth files
- [ ] if a JIRA ticket exists, [make sure it is resolved properly](https://github.com/spacetelescope/romancal/wiki/How-to-resolve-JIRA-issues)

<details><summary>news fragment change types...</summary>

  - ``changes/<PR#>.general.rst``: infrastructure or miscellaneous change
  - ``changes/<PR#>.docs.rst``
  - ``changes/<PR#>.stpipe.rst``
  - ``changes/<PR#>.associations.rst``
  - ``changes/<PR#>.scripts.rst``
  - ``changes/<PR#>.mosaic_pipeline.rst``
  - ``changes/<PR#>.patch_match.rst``

  ## steps
  - ``changes/<PR#>.dq_init.rst``
  - ``changes/<PR#>.saturation.rst``
  - ``changes/<PR#>.refpix.rst``
  - ``changes/<PR#>.linearity.rst``
  - ``changes/<PR#>.dark_current.rst``
  - ``changes/<PR#>.jump_detection.rst``
  - ``changes/<PR#>.ramp_fitting.rst``
  - ``changes/<PR#>.assign_wcs.rst``
  - ``changes/<PR#>.flatfield.rst``
  - ``changes/<PR#>.photom.rst``
  - ``changes/<PR#>.flux.rst``
  - ``changes/<PR#>.source_detection.rst``
  - ``changes/<PR#>.tweakreg.rst``
  - ``changes/<PR#>.skymatch.rst``
  - ``changes/<PR#>.outlier_detection.rst``
  - ``changes/<PR#>.resample.rst``
  - ``changes/<PR#>.source_catalog.rst``
</details>

### CODE REVIEW

**Review by schlafly — COMMENTED** [review]
A few comments inline, thanks!

File: romancal/resample/resample.py [hunk 1]
```diff
@@ -921,12 +919,7 @@ def populate_mosaic_basic(
     output_model.meta.basic.program = (
         input_meta[0].observation.program
         if len({x.observation.program for x in input_meta}) == 1
-        else "-1"
-    )
-    output_model.meta.basic.survey = (
-        input_meta[0].observation.survey
-        if len({x.observation.survey for x in input_meta}) == 1
-        else "MULTIPLE"
```
[the remark this answers is not in the document]
Comment (schlafly) [reply romancal/resample/resample.py:929]: i.e., fine to leave this as is for now, thanks.

File: romancal/ramp_fitting/ramp_fit_step.py [hunk 2]
```diff
@@ -88,7 +86,7 @@ def ols_cas22(self, input_model, readnoise_model, gain_model, dark_model):
         out_model : ImageModel
             Model containing a count-rate image.
         """
-        use_jump = self.use_ramp_jump_detection
```
[the remark this answers is not in the document]
Comment (mairanteodoro) [reply romancal/ramp_fitting/ramp_fit_step.py]: Reverted.

File: romancal/resample/resample.py [hunk 3]
```diff
@@ -181,10 +181,10 @@ def __init__(
...
             self.blank_output.meta.wcs = self.output_wcs
             gwcs_into_l3(self.blank_output, self.output_wcs)
-            self.blank_output.cal_logs = stnode.CalLogs()
-            self.blank_output["individual_image_cal_logs"] = [
-                model.cal_logs for model in models
-            ]
+            # self.blank_output.cal_logs = stnode.CalLogs()
+            # self.blank_output["individual_image_cal_logs"] = [
+            #     model.cal_logs for model in models
+            # ]
```
[the remark this answers is not in the document]
Comment (mairanteodoro) [reply romancal/resample/resample.py]: Done.

File: romancal/resample/resample_step.py [hunk 4]
```diff
@@ -156,7 +156,7 @@ def _final_updates(self, model, input_models, kwargs):
         # calculate the actual value of pixel_scale_ratio based on pixel_scale
         # because source_catalog uses this value from the header.
         model.meta.resample.pixel_scale_ratio = (
-            self.pixel_scale / np.sqrt(model.meta.photometry.pixelarea_arcsecsq)
+            self.pixel_scale / np.sqrt(model.meta.photometry.pixel_area * 4.254517e10)
```
[the remark this answers is not in the document]
Comment (mairanteodoro) [reply romancal/resample/resample_step.py]: Done.

**Review by schlafly — COMMENTED** [review]
This looks good to me.  A few very minor comments inline.

**Review by PaulHuwe — APPROVED** [review]
LGTM

**Review by schlafly — APPROVED** [review]
Thanks, looks good!

**Pull request merged**